contracts/NXMToken.sol

```diff
@@ -0,0 +1,336 @@
+        public
+        returns (bool)
+    {
+        require(spender != address(0));
+
+        _allowed[msg.sender][spender] = (
+        _allowed[msg.sender][spender].add(addedValue));
+        emit Approval(msg.sender, spender, _allowed[msg.sender][spender]);
+        return true;
+    }
+
+    /**
+    * @dev Decrease the amount of tokens that an owner allowed to a spender.
+    * approve should be called when allowed_[_spender] == 0. To decrement
+    * allowed value is better to use this function to avoid 2 calls (and wait until
+    * the first transaction is mined)
+    * From MonolithDAO Token.sol
+    * @param spender The address which will spend the funds.
+    * @param subtractedValue The amount of tokens to decrease the allowance by.
+    */
+    function decreaseAllowance(
+        address spender,
+        uint256 subtractedValue
+    )
+        public
+        returns (bool)
+    {
+        require(spender != address(0));
+
+        _allowed[msg.sender][spender] = (
+        _allowed[msg.sender][spender].sub(subtractedValue));
+        emit Approval(msg.sender, spender, _allowed[msg.sender][spender]);
+        return true;
+    }
+
+    /**
+    * @dev Adds a user to whitelist
+    * @param _member address to add to whitelist
+    */
+    function addToWhiteList(address _member) public onlyOperator returns (bool) {
+        whiteListed[_member] = true;
+        emit WhiteListed(_member);
+        return true;
+    }
+
+    /**
+    * @dev removes a user from whitelist
+    * @param _member address to remove from whitelist
+    */
+    function removeFromWhiteList(address _member) public onlyOperator returns (bool) {
+        whiteListed[_member] = false;
+        emit BlackListed(_member);
+        return true;
+    }
+
+    /**
+    * @dev burns an amount of the tokens of the message sender
+    * account.
+    * @param amount The amount that will be burnt.
+    */
+    function burn(uint256 amount) public returns (bool) {
+        _burn(msg.sender, amount);
+        return true;
+    }
+
+    /**
+    * @dev Burns a specific amount of tokens from the target address and decrements allowance
+    * @param from address The address which you want to send tokens from
+    * @param value uint256 The amount of token to be burned
+    */
+    function burnFrom(address from, uint256 value) public {
+        _burnFrom(from, value);
+    }
+
+    /**
+    * @dev function that mints an amount of the token and assigns it to
+    * an account.
+    * @param account The account that will receive the created tokens.
+    * @param amount The amount that will be created.
+    */
+    function mint(address account, uint256 amount) public onlyOperator {
+        _mint(account, amount);
+    }
+
+    /**
+    * @dev Transfer token for a specified address
+    * @param to The address to transfer to.
+    * @param value The amount to be transferred.
+    */
+    function transfer(address to, uint256 value) public canTransfer(to) returns (bool) {
```

Comment:
Need to add modifier for checking emergency pause. There are other functions which require the check as well for eg transfer from, mint, burn etc. Please refer the older contracts here.

Reply:
Got it, working on it